### PR TITLE
Mac support via #defines, some #ifdeffing

### PR DESCRIPTION
**Pull request by darknoon**
This is preliminary, intented for commentary / review of basic approach.

A lot of the CKM classes are non-final, and some merging of API / functionality should really occur between iOS and mac text labels, buttons, etc.

### CODE REVIEW

File: ComponentKit.podspec [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
   s.version = "0.12"
-  s.summary = "A React-inspired view framework for iOS"
+  s.summary = "A React-inspired view framework for iOS and OS X"
   s.homepage = "https://componentkit.org"
-  s.authors = 'adamjernst@fb.com'
+  s.authors = 'adamjernst@fb.com', 'andrewpouliot@fb.com', 'beefon@fb.com',
   s.license = 'BSD'
   s.source = {
-    :git => "https://github.com/facebook/ComponentKit.git",
+    :git => "https://github.com/darknoon/ComponentKit-Mac.git",
```
Comment (adamjernst) [inline ComponentKit.podspec]: Oops, revert

File: ComponentKit/Accessibility/CKComponentAccessibility.mm [hunk 2]
```diff
@@ -24,7 +24,11 @@ static CKViewComponentAttributeValueMap ViewAttributesFromAccessibilityContext(c
     accessibilityAttributes[@selector(setAccessibilityIdentifier:)] = accessibilityContext.accessibilityIdentifier;
   }
   if (accessibilityContext.isAccessibilityElement) {
+#if TARGET_OS_IPHONE
     accessibilityAttributes[@selector(setIsAccessibilityElement:)] = accessibilityContext.isAccessibilityElement;
+#else
```
Comment (adamjernst) [inline ComponentKit/Accessibility/CKComponentAccessibility.mm]: I might prefer else-if's and then an #else with #error unrecognized platform... but that's probably overkill

Comment (darknoon) [inline ComponentKit/Accessibility/CKComponentAccessibility.mm]: That's reasonable. Apple's decision to make iOS also `TARGET_OS_MAC` defines the order, but I can make the else simply `TARGET_OS_MAC`.

**Comment by adamjernst** [comment]
Looks good to me in general; closing until it builds properly, or I have time to fix builds.

**Pull request closed without merging**